scripts/combine-profile-information.py

```diff
@@ -0,0 +1,128 @@
+    numeric = "".join(map(str, range(10))) + "."
+    unit_start_position = next(
+        (i for i, c in enumerate(number_string) if c not in numeric), len(number_string)
+    )
+    number = float(number_string[:unit_start_position])
+    unit = number_string[unit_start_position:]
+    if number.is_integer():
+        number = int(number)
+    return number, unit
+
+
+PRIMARY_KEY = "pass name"
+summary_non_counter_fields = set()
+summary_counter_fields = set()
+
+for rows in map(get_csv_rows, get_input_csv_paths()):
+    curr_fields = set(rows[0])
+    curr_fields.remove(PRIMARY_KEY)
+    if "counters" in curr_fields:
+        curr_fields.remove("counters")
+        for row in rows:
+            summary_counter_fields.update(parse_counters(row["counters"]))
+    summary_non_counter_fields.update(curr_fields)
+
+# CR mitom: Not supporting memory for now
+summary_non_counter_fields -= {
+    "alloc",
+    "top-heap",
+    "absolute-top-heap",
+}
+
+# Ensure consistent ordering of summary fields
+field_collections = {PRIMARY_KEY}, summary_non_counter_fields, summary_counter_fields
+SUMMARY_FIELD_NAMES = sum(map(lambda fields: sorted(fields), field_collections), [])
+
+units = {}
+for field in summary_non_counter_fields:
+    first_row = get_csv_rows(next(get_input_csv_paths()))[0]
+    _, units[field] = split_into_number_and_unit(first_row[field])
+
+
+def strip_units(row: CsvRow) -> CsvRow:
+    return {k: (v[: -len(units[k])] if k in units else v) for k, v in row.items()}
+
+
+def row_summary(row: CsvRow) -> SummaryRow:
+    full_summary = {**strip_units(row), **parse_counters(row["counters"])}
+    return {k: v for k, v in full_summary.items() if k in SUMMARY_FIELD_NAMES}
+
+
+def csv_to_summaries(rows: CsvRows) -> Iterable[SummaryRow]:
+    counter_rows = [row for row in rows if row["counters"]]
+    return map(row_summary, counter_rows)
+
+
+def split_into_number_and_unit(number_string: str) -> (str, Union[int, float]):
+    number_chars = "".join(str(i) for i in range(10)) + "."
+    non_numeric = (i for i, c in enumerate(number_string) if c not in number_chars)
+    suffix_start = next(non_numeric, len(number_string))
+    number, unit = float(number_string[:suffix_start]), number_string[suffix_start:]
+    if number.is_integer():
+        number = int(number)
+    return number, unit
+
+
+def process_csv(csv_path: Path, summary_csv_writer: csv.DictWriter) -> None:
+    for summary in csv_to_summaries(get_csv_rows(csv_path)):
+        summary_csv_writer.writerow(summary)
+
+
+with open(SUMMARY_PATH, "w", newline="") as summary_csv_file:
+    summary_csv_writer = csv.DictWriter(
+        summary_csv_file, fieldnames=SUMMARY_FIELD_NAMES
+    )
+    summary_csv_writer.writeheader()
+    # Add units as second header (can use header=[0,1] in pandas.read_csv)
+    summary_csv_writer.writerow(units)
+
+    for curr_csv_path in get_input_csv_paths():
+        process_csv(curr_csv_path, summary_csv_writer)
+        curr_csv_path.unlink()
```

Comment:
(Not sure about that, eventually we
may want to apply the script several
times to a dump in order to extract
different values.)

Reply:
My thought process was removing it here provides some safety in case someone accidentally changes the dump directory as it only removes CSV files. However, if we use the `_profile` folder and check that it is empty as described below, we can remove these files in `profile-compiler-build.sh`.